user_sync/connector/directory_okta.py

```diff
@@ -0,0 +1,290 @@
+        try:
+            results = self.groups_client.get_groups(query=group_filter_format.format(group=group))
+        except Exception as e:
+            self.logger.warning("Unable to query group")
+            raise user_sync.error.AssertionException(repr(e))
+
+        if results is None:
+            raise user_sync.error.AssertionException("Group not found for: %s" % group)
+        else:
+            for result in results:
+                if result.profile.name == group:
+                    return result
+
+        return None
+
+    def iter_group_members(self, group, filter_string, extended_attributes):
+        """
+        :type group: str
+        :type extended_attributes: list
+        :rtype iterator(str, str)
+        """
+
+        user_attribute_names = ["firstName", "lastName", "login", "email", "countryCode"]
+        extended_attributes = list(set(extended_attributes) - set(user_attribute_names))
+        user_attribute_names.extend(extended_attributes)
+
+        res_group = self.find_group(group)
+        if res_group:
+            try:
+                attr_dict = OKTAValueFormatter.get_extended_attribute_dict(user_attribute_names)
+                members = self.groups_client.get_group_all_users(res_group.id, attr_dict)
+            except Exception as e:
+                self.logger.warning("Unable to get_group_users")
+                raise user_sync.error.AssertionException(repr(e))
+            #Filtering users based all_users_filter query in config
+            if filter_string:
+                members = self.filter_user(members,filter_string)
+            for member in members:
+                profile = member.profile
+                if not profile.email:
+                    # if (last_attribute_name != None):
+                    self.logger.warn('No email attribute for login: %s', profile.login)
+                    continue
+
+                user = self.convert_user(member, extended_attributes)
+                if not user:
+                    continue
+                yield (user)
+        else:
+            self.logger.warning("No group found for: %s", group)
+
+    def convert_user(self, record, extended_attributes):
+        profile = record.profile
+
+        source_attributes = {}
+        user = user_sync.connector.helper.create_blank_user()
+
+        source_attributes['id'] = user['uid'] = record.id
+        source_attributes['email'] = user['email'] = profile.email
+
+        source_attributes['identity_type'] = user_identity_type = self.user_identity_type
+        if not user_identity_type:
+            user['identity_type'] = self.user_identity_type
+        else:
+            try:
+                user['identity_type'] = user_sync.identity_type.parse_identity_type(user_identity_type)
+            except user_sync.error.AssertionException as e:
+                self.logger.warning('Skipping user %s: %s', profile.login, e.message)
```

Comment:
change `e.message` to `e`.  That will cause %s to invoke repr(e) rather than just the message component.  I notice this also needs to be fixed in the LDAP module.  I'll take care of that.